### ISSUE TITLE
(Feature):  [Windows] Upgrade to OpenEXR 3.1

### ISSUE THREAD

**Issue by 360highdynamicrangehdr** [NatronGitHub/Natron#870]
### Make sure to follow our issue report guidelines

- [X] I have read the [feature request guidelines](https://hackmd.io/@natron-dev-awesome/B1SW6Hbau)
- [X] I have followed the [contributing guidelines](https://github.com/NatronGitHub/Natron/blob/RB-2.4/CONTRIBUTING.md) to the best of my understanding
- [X] This feature is not on the [issue tracker](https://github.com/NatronGitHub/Natron/issues?q=is%3Abug+type%3Afeature+type%3Aidea) and is not a duplicate

### Provide a description of your feature request

Upgrade OpenEXR 3.1.3 or later.

a compression type `dwaa` and `dwab`, `zip` and `zips`

- [x] Zip compression level
- [x] Dwa compression level

see https://aras-p.info/blog/2021/08/05/EXR-Zip-compression-levels/

### Is this a realistically implementable feature?

- [X] My feature is realistically implementable
- [ ] My feature is NOT realistically implementable

### Can you contribute in creating this feature?

- [ ] I am unable, or not willing to contribute to creating this feature
- [ ] I am able and willing to contribute to the creation of this feature

**Comment by devernay** [comment]
My guess is that this is a windows-specific issue, since both macOS and Linux binaries ship with OpenEXR 3.1.5, and have zips, dwaa and dwab. @rodlie

**Comment by rodlie** [comment]
OpenEXR 3 was not available for mingw when I last updated the Windows SDK, it's now available. Will update the SDK sometime this week.

**Comment by devernay** [comment]
isn't this fixed by https://github.com/NatronGitHub/Natron/commit/a0a10a1a01d86d3d455cc64098f217fa973c3301 ?

**Comment by rodlie** [comment]
Yes, in theory. I was not able to run a full Natron build since gmic.eu was down when I did the change. Will run a new build today.

**Comment by rodlie** [comment]
So.... curl fails when connecting to gmic.eu, the domain is not down.
```
2023-04-16 14:41:38 Building openfx-gmic...
Info: build openfx-gmic using make...
curl -L -s -S -o gmic_2.8.4.tar.gz http://gmic.eu/files/source/gmic_2.8.4.tar.gz
curl -L -s -S -o gmic_libc.h https://raw.githubusercontent.com/dtschump/gmic-community/6a9d0e344162af3376aee7b652721d972a070e6b/libcgmic/gmic_libc.h
curl -L -s -S -o gmic_stdlib_gmic.h https://raw.githubusercontent.com/dtschump/gmic-community/6a9d0e344162af3376aee7b652721d972a070e6b/libcgmic/gmic_stdlib_gmic.h
curl: (6) Could not resolve host: gmic.eu
make: *** [Makefile:45: gmic_2.8.4.tar.gz] Error 6
make: *** Waiting for unfinished jobs....
```

But downloading it manually from the terminal works just fine:
```
$ wget http://gmic.eu/files/source/gmic_2.8.4.tar.gz
--2023-04-16 16:43:16--  http://gmic.eu/files/source/gmic_2.8.4.tar.gz
Resolving gmic.eu (gmic.eu)... 213.186.33.40
Connecting to gmic.eu (gmic.eu)|213.186.33.40|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 6182300 (5.9M) [application/x-gzip]
Saving to: ‘gmic_2.8.4.tar.gz’

gmic_2.8.4.tar.gz               100%[====================================================>]   5.90M  2.88MB/s    in 2.0s

2023-04-16 16:43:18 (2.88 MB/s) - ‘gmic_2.8.4.tar.gz’ saved [6182300/6182300]
```

The curl command from the Makefile also works if I run it manually
```
$ curl -L -s -S -o gmic_2.8.4.tar.gz http://gmic.eu/files/source/gmic_2.8.4.tar.gz
```

curl has not been changed, so why this all of the sudden fail I don't know. Will check a bit later today.

**Comment by devernay** [comment]
my guess is your DNS requires some certificate that is not in your current version. Try with option `-k` or `--insecure` of curl, and if it works it means you probably have to upgrade curl and its ca-certificates

**Comment by devernay** [comment]
And also, probably a different curl is executed from the command-line, or with a different environment that lets it fetch its certs from another place

**Comment by devernay** [comment]
fixed by a0a10a1a01d86d3d455cc64098f217fa973c3301 - will be in 2.5.1 and 2.6.0

2.6.0 test builds available for windows: https://github.com/NatronGitHub/Natron/releases/download/v2.6.0-alpha1/Natron-RB-2.6-202305242302-b9a9b81-Windows-x86_64.zip
